backend/src/crash/main.py

```diff
@@ -194,15 +223,28 @@ async def websocket_endpoint(websocket: WebSocket):
     except WebSocketDisconnect:
 
         manager.disconnect(websocket)
-        new_player_list = list(current_players.values())
-        new_player_list.remove(current_players[websocket])
+        new_player_list = [player.name for player in current_players.values()]
+        new_player_list.remove(current_players[websocket].name)
+        print("new lobby list : ", new_player_list)
         await manager.broadcast_lobby(
             {
                 "type": "leave",
-                "target": current_players[websocket],
+                "target": current_players[websocket].name,
                 "lobby": new_player_list,
                 "bids": game.get_bids(current_players),
                 "cash_vaults": game.get_cash_vaults(current_players),
             }
         )
+        # Update the information in the db
+        players_collection.update_one(
+            {"name": current_players[websocket].name},
+            {
+                "$set": {"cash": current_players[websocket].cash},
+                "$push": {
+                    "bid_history": current_players[websocket].bid_history,
+                    "gain_history": current_players[websocket].gain_history,
+                    "mult_history": current_players[websocket].mult_history,
+                },
+            },
+        )
```

Comment:
I like the idea of syncing up with the db when the client disconnects. It avoids unnecessary writes to the db, and should avoid data loss in normal operation.
The only issue I see is in case of crash, where any data between client join and server crash will be lost. 
An alternative solution is to write data as it changes, i.e. a round ends with a bid placed during the round -> write that new info.

Not a big deal for now though.